check_for_update.go

```diff
@@ -50,12 +50,13 @@ func getLatestVersion() (*semver.Version, error) {
 	if err != nil || time.Since(modified) > 24*time.Hour {
 		versionOnline, err := getLatestVersionFromURL()
 		if err == nil {
-			version = versionOnline
-			ioutil.WriteFile(versionCacheFilename, []byte(version.String()), 0600)
+			ioutil.WriteFile(versionCacheFilename, []byte(versionOnline.String()), 0600)
 		}
+
+		return versionOnline, err
 	}
 
-	return version, nil
+	return version, err
```

Comment:
Well, now (after the fix) we will never reach this point if `getLatestVersionFromCache()` returns an err != nil.
It was possible before, because this return statement was used every time, even when both `getLatestVersionFromCache()` and `getLatestVersionFromURL()` failed. In that case, the hard-coded `nil` "shadowed" the error from `getLatestVersionFromURL()`.